### ISSUE TITLE
[java] UnusedAssignment FP in try/catch

### ISSUE THREAD

**Issue by kris-scheibe** [pmd/pmd#2684]
**Affects PMD Version:** 6.26.0 (and 6.27.0-SNAPSHOT)

**Rule:** [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment)

**Description:**
A method-call within a try-block might fail and its result shouldn't be treated as a guaranteed assignment.

**Code Sample demonstrating the issue:**
```
public class Foo {

    public int foo() {
        int a = 0; // <--- false-positive UnusedAssignment
        try (Reader r = new StringReader("")) {
            a = r.read(); // might assign or fail
            a = r.read(); // might assign or fail
        } catch (IOException e) {
        }
        return a;
    }

}
```

**Expected outcome:**

*   Both the first and second read might fail. All assignments to the variable a are useful, depending on whether one of the read()'s fails -> false-positive

**Running PMD through:** *Other*
